### ISSUE TITLE
No background blur for qt popups

### ISSUE THREAD

**Issue by joelmathewthomas** [hyprwm/Hyprland#6453]
### Bug or Regression?

Bug

### System Info and Version

<details> <summary>System/Version info</summary>

```shHyprland, built from branch  at commit 4cdddcfe466cb21db81af0ac39e51cc15f574da9  (cursor: minor fixes for unhiding surfaces).
Date: Tue May 14 21:15:12 2024
Tag: v0.40.0-80-g4cdddcfe, commits: 4686

flags: (if any)


System Information:
System name: Linux
Node name: nitro5
Release: 6.9.1-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Fri, 17 May 2024 16:56:38 +0000


GPU information: 
0000:00:02.0 VGA compatible controller [0300]: Intel Corporation TigerLake-H GT1 [UHD Graphics] [8086:9a60] (rev 01) (prog-if 00 [VGA controller])
0000:01:00.0 VGA compatible controller [0300]: NVIDIA Corporation GA107M [GeForce RTX 3050 Ti Mobile] [10de:25a0] (rev a1) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```

</details>

### Description

Popups for qt apps are not displayed properly.

With option `popups=true` in the `blur` section of Hyprland config, it displays it with no background.

![2024_06_13_00_18_05](https://github.com/hyprwm/Hyprland/assets/90510078/66fbff28-de5d-4cf5-97be-1bb70737de76)

But for some reason when i trigger a popup while rofi is open, it displays it properly with intended blur.

https://github.com/hyprwm/Hyprland/assets/90510078/446e4ed0-bdeb-49b4-990a-1e1526c17a42

Now if i turn off blur for popups , then it displays it properly with no blur, but not so readable, and not how it was intended to be.

https://github.com/hyprwm/Hyprland/assets/90510078/8c976212-8f07-401a-b191-a22bdabef3fe

### How to reproduce

Try to display some popups with the configuration as:

`popups = true` for `blur` section.

### Crash reports, logs, images, videos

Crash Logs reported by dolphin during this process is

`qt.qpa.wayland: Creating a popup with a parent, QWidgetWindow(0x5d330f12efb0, name="Dolphin#1Window") which does not match the current topmost grabbing popup, QWidgetWindow(0x5d330f433f30, name="DolphinContextMenuClassWindow") With some shell surface protocols, this is not allowed. The wayland QPA plugin is currently handling it by setting the parent to the topmost grabbing popup. Note, however, that this may cause positioning errors and popups closing unxpectedly. Please fix the transient parent of the popup.`

**Comment by vaxerski** [comment]
maybe because it's opaque??

**Comment by joelmathewthomas** [comment]
> maybe because it's opaque??

No, because like I said, it displays properly when rofi is triggered.
https://github.com/hyprwm/Hyprland/assets/90510078/84069249-78ae-4e81-ae84-149411c026f3

**Comment by vaxerski** [comment]
mybad

**Comment by vaxerski** [comment]
is this only qt?

**Comment by joelmathewthomas** [comment]
> is this only qt?

give me a minute. let me do a quick check.

**Comment by vaxerski** [comment]
it is blurred on my end, I just noticed you're on an old version. Try updating.

**Comment by joelmathewthomas** [comment]
No, it happens with gtk as well.

this is with thunar.

https://github.com/hyprwm/Hyprland/assets/90510078/759fb211-2260-457d-af3f-445db01402de

**Comment by joelmathewthomas** [comment]
> it is blurred on my end, I just noticed you're on an old version. Try updating.

Guess I will update you, after i update Hyprland.

**Comment by joelmathewthomas** [comment]
> it is blurred on my end, I just noticed you're on an old version. Try updating.

I'd have to say no change.

Also, now i get more visual artifacts.

Earlier, while opening and closing rofi , only the popups had some sort of glitching.

Now the main dolphin window also has the glitching.


https://github.com/hyprwm/Hyprland/assets/90510078/1effb66e-12db-440c-b5ce-992af00fcbbd



I am on commit `38132ffaf53c843a3ed03be5b305702fde8f5a49`.


My Hyprland config is
[hyprland.conf.txt](https://github.com/user-attachments/files/15810073/hyprland.conf.txt)

**Comment by vaxerski** [comment]
looks funky

**Comment by vaxerski** [comment]
can't repro still

what if you set `opengl:force_introspection` to true?

**Comment by joelmathewthomas** [comment]
> can't repro still
> 
> what if you set `opengl:force_introspection` to true?

I'm sorry. I don't understand.
How do i do this.

**Comment by vaxerski** [comment]
```ini
opengl {
    force_introspection = true
}
```
...

**Comment by joelmathewthomas** [comment]
> ```ini
> opengl {
>     force_introspection = true
> }
> ```
> 
> ...

Yes, now it works fine.

**Comment by vaxerski** [comment]
funky opengl stuff

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
